nexus/src/app/sagas/instance_create.rs

```diff
@@ -281,36 +276,55 @@ impl NexusSaga for SagaInstanceCreate {
             ));
             subsaga_append(
                 "external_ip",
-                subsaga_builder,
+                subsaga_builder.build()?,
                 &mut builder,
                 repeat_params,
                 i,
             )?;
         }
 
-        // See the comment above where we add nodes for creating NICs.  We use
-        // the same pattern here.
-        for i in 0..(MAX_DISKS_PER_INSTANCE as usize) {
-            let disk_params =
-                DiskParams { saga_params: params.clone(), which: i };
+        // Appends the disk create saga as a subsaga directly to the instance create builder.
+        for (i, disk) in params.create_params.disks.iter().enumerate() {
```

Comment:
Nice; stoked to see we are now only generating nodes for the portion of the graph we need.